src/codegeneration/ModuleTransformer.js

```diff
@@ -107,7 +107,7 @@ export class ModuleTransformer extends TempVarTransformer {
 
     statements = this.wrapModule(this.moduleProlog().concat(statements));
 
-    return new Script(tree.location, statements);
+    return new Script(tree.location, statements, tree.moduleName);
```

Comment:
I wonder if we should be passing `null` here.  We've just stored `tree.moduleName` in `this.moduleName`  and we'll eventually use that value to generate code.  The `Script()` we return here is more of an anonymous statement list.